src/Transport/DockerComposeTransport.php

```diff
@@ -0,0 +1,92 @@
+<?php
+
+namespace Consolidation\SiteProcess\Transport;
+
+use Drush\Drush;
+use Psr\Log\LoggerInterface;
+use Robo\Common\IO;
+use Symfony\Component\Console\Style\OutputStyle;
+use Symfony\Component\Process\Process;
+use Consolidation\SiteProcess\Util\RealtimeOutputHandler;
+use Consolidation\SiteProcess\Util\Escape;
+use Symfony\Component\Console\Output\ConsoleOutputInterface;
+use Consolidation\SiteAlias\AliasRecord;
+use Consolidation\SiteProcess\Util\Shell;
+
+/**
+ * DockerComposeTransport knows how to wrap a command such that it executes on a Docker Compose service.
+ */
+class DockerComposeTransport implements TransportInterface
+{
+    protected $tty;
+    protected $siteAlias;
+    protected $cd;
+
+    public function __construct(AliasRecord $siteAlias)
+    {
+        $this->siteAlias = $siteAlias;
+    }
+
+    /**
+     * inheritdoc
+     */
+    public function configure(Process $process)
+    {
+        $this->tty = $process->isTty();
+    }
+
+    /**
+     * inheritdoc
+     */
+    public function wrap($args)
+    {
+        $transport = ['docker-compose', 'exec'];
+        $transportOptions = $this->getTransportOptions();
+        $commandToExecute = $this->getCommandToExecute($args);
+
+        return array_merge(
+            $transport,
+            $transportOptions,
+            $commandToExecute
+        );
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function addChdir($cd, $args)
+    {
+        $this->cd = $cd;
+    }
+
+    /**
+     * getTransportOptions returns the transport options for the tranport
+     * mechanism itself
+     */
+    protected function getTransportOptions()
+    {
+        $transportOptions[] = $this->siteAlias->get('service');
+        if ($options = $this->siteAlias->get('exec.options')) {
+            array_unshift($transportOptions, Shell::preEscaped($options));
```

Comment:
Do `exec.options` need to be pre-escaped? Seems like it would be better to escape them here, and include them raw in the alias file, as usual.

Reply:
I guess this is analogous to `ssh.options`?

Maybe `exec.options` is too generic; if this setting is supposed to be specific to docker-composer exec, maybe it should be in the `docker` namespace.

Reply:
If `exec.option` works like `ssh.options`, then pre-escaping is correct. I think that perhaps this code should go back to the way it is, perhaps with different namespacing.

To me, it feels like `docker-compose exec` vs `docker exec` is an implementation detail. If that's the case, and no one would make a `docker` transport once the `docker-compose` transport is available, then I think I'd prefer the namespace to simply be `docker`.

Unsure about `service`. Need to contemplate the usage in Drush a bit.